### PR TITLE
Added user & order mailer

### PR DESCRIPTION
**Pull request by gsnovak**
@fiestacasey @austinkahly @ksolkowski

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export SENDGRID_PASSWORD=fd7yj2lc5084
```
Comment (fiestacasey) [inline .env:1]: omg. git ignore this.

File: app/views/order_mailer/send_cancellation_email.html.haml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+%h1
+  Order # #{order.id} has been canceled
+  %p Thank you for buying our expensive paper products!
```
Comment (fiestacasey) [inline app/views/order_mailer/send_cancellation_email.html.haml]: lol

**Review by fiestacasey — APPROVED** [review]
Approved as is but... Sending an email relies on a third party service being up and fast. So it should be a background process and never affect the customer waiting on the browser. 

Next project; push these emails to a sidekiq queue.

File: app/mailers/order_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class OrderMailer < ActionMailer::Base
+  default :from => 'graham@wantable.com'
+
+  def send_placement_email(order)
+    @order = order
+    mail( :to => order.user.email,
+    :subject => 'Your order has been placed.' )
```
Comment (ksolkowski) [inline app/mailers/order_mailer.rb]: use ```:``` instead of ```=>``` NO ROCKETS!

Comment (ksolkowski) [inline app/mailers/order_mailer.rb]: can just put it all in one line.

File: app/mailers/order_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  def send_cancellation_email(order)
+    @order = order
+    mail( to: order.user.email,
```
Comment (ksolkowski) [inline app/mailers/order_mailer.rb]: can put it all one line

File: app/views/order_mailer/send_placement_email.html.haml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+%h1
+  Order # #{order.id} has been placed
```
Comment (ksolkowski) [inline app/views/order_mailer/send_placement_email.html.haml]: ```#{@order.id}```

File: config/routes.rb [hunk 6]
```diff
@@ -2,17 +2,18 @@
...
   namespace :api do
     namespace :v1 do
-      resources :orders do
-         get 'cart', on: :collection, to: 'orders#cart'
+      resources :order do
+         get 'cart', on: :collection, to: 'order#cart'
+         put :update
       end
-      resources :products
-      resources :order_items do
```
Comment (ksolkowski) [inline config/routes.rb]: shouldn't it be plural?

File: test/mailers/order_mailer.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      email.deliver_now
+    end
+
+    assert_equal ['me@example.com'], email.from
+    assert_equal ['friend@example.com'], email.to
+    assert_equal 'Your order has been placed.', email.subject
+  end
+
+  test "cancelation" do
+    email = OrderMailer.send_cancellation_email('me@example.com',
```
Comment (ksolkowski) [inline test/mailers/order_mailer.rb]: ```OrderMailer.send_cancellation_email```  only takes 1 argument i thought.... Unless action mailer changes things....

**Review by ksolkowski — CHANGES_REQUESTED** [review]

**Pull request merged**